schema: TestRevertingJobsOnDatabasesAndSchemas/failed_due_to_injected_error/drop_database_cascade flaky

[TestRevertingJobsOnDatabasesAndSchemas/failed_due_to_injected_error/drop_database_cascade](https://teamcity.cockroachdb.com//viewLog.html?buildId=4457751&buildTypeId=Cockroach_MergedExtendedCi#testNameId-1013794384827983388) is flaky, see the test history: https://teamcity.cockroachdb.com/project.html?projectId=Cockroach_UnitTests&testNameId=-1013794384827983388&tab=testDetails 

Going to skip for now.

Removing test in https://github.com/cockroachdb/cockroach/pull/77049

Jira issue: CRDB-13384

I'm reopening this issue because I witnessed this test fail on a recent CI build: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/5966731?showRootCauses=true&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
------- Stdout: -------
=== RUN   TestRevertingJobsOnDatabasesAndSchemas/failed_due_to_injected_error/drop_schema
    sql_runner.go:101:
          Error Trace:  sql_runner.go:114
                              sql_runner.go:284
                              schema_changer_test.go:6931
          Error:        Received unexpected error:
                        query 'SELECT num_runs > 3 FROM crdb_internal.jobs WHERE status = 'reverting' AND description ~ '^DROP SCHEMA db_drop_schema.sc$'': expected:
                        (1) attached stack trace
                          -- stack trace:
                          | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:287
                          | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:69
                          | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:207
                          | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
                          | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:114
                          | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:284
                          | github.com/cockroachdb/cockroach/pkg/sql_test.TestRevertingJobsOnDatabasesAndSchemas.func1.2
                          |   /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer_test.go:6931
                          | testing.tRunner
                          |   /usr/local/go/src/testing/testing.go:1439
                          | runtime.goexit
                          |   /usr/local/go/src/runtime/asm_amd64.s:1571
                        Wraps: (2) query 'SELECT num_runs > 3 FROM crdb_internal.jobs WHERE status = 'reverting' AND description ~ '^DROP SCHEMA db_drop_schema.sc$'': expected:
                          | true
                          | got:
                          |
                        Error types: (1) *withstack.withStack (2) *errutil.leafError
        --- FAIL: TestRevertingJobsOnDatabasesAndSchemas/failed_due_to_injected_error/drop_schema (45.38s)
```

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -6923,8 +6923,7 @@ func TestRevertingJobsOnDatabasesAndSchemas(t *testing.T) {
 				go func(scStmt string) {
 					// This transaction will not return until the server is shutdown. Therefore,
 					// we run it in a separate goroutine and don't check the returned error.
-					sqlDB.Exec(t, `SET use_declarative_schema_changer = 'off'`)
-					_, _ = db.Exec(scStmt)
+					_, _ = db.Exec(`SET use_declarative_schema_changer = 'off'; ` + scStmt)
 				}(tc.scStmt)
 				// Verify that the job is in retry state while reverting.
 				const query = `SELECT num_runs > 3 FROM crdb_internal.jobs WHERE status = '` + string(jobs.StatusReverting) + `' AND description ~ '%s'`

